sdcm/gemini_thread.py

```diff
@@ -66,50 +66,105 @@ def __init__(self, test_cluster, oracle_cluster, loaders, stress_cmd, timeout=No
         super().__init__(loader_set=loaders, stress_cmd=stress_cmd, timeout=timeout, params=params)
         self.test_cluster = test_cluster
         self.oracle_cluster = oracle_cluster
-        self._gemini_result_file = None
         self.gemini_commands = []
-        self.gemini_request_timeout = 180
-        self.gemini_connect_timeout = 120
-
-    @property
-    def gemini_result_file(self):
-        if not self._gemini_result_file:
-            self._gemini_result_file = os.path.join("/", "gemini_result_{}.log".format(uuid.uuid4()))
-        return self._gemini_result_file
+        self.unique_id = uuid.uuid4()
+        self.gemini_default_flags = {
+            "level": "info",
+            "request-timeout": "60s",
+            "connect-timeout": "60s",
+            "consistency": "QUORUM",
+            "async-objects-stabilization-backoff": "1s",
+            "async-objects-stabilization-attempts": 10,
+            "max-mutation-retries-backoff": "1s",
+            "max-mutation-retries": 10,
+            "dataset-size": "large",
+            "oracle-host-selection-policy": "token-aware",
+            "test-host-selection-policy": "token-aware",
+            "drop-schema": "true",
+            "cql-features": "normal",
+            "fail-fast": "true",
+            "materialized-views": "false",
+            "use-server-timestamps": "true",
+            "use-lwt": "false",
+            "use-counters": "false",
+            "max-tables": 1,
+            "max-columns": 16,
+            "min-columns": 8,
+            "max-partition-keys": 6,
+            "min-partition-keys": 2,
+            "max-clustering-keys": 4,
+            "min-clustering-keys": 2,
+            "partition-key-distribution": "normal",  # Distribution for hitting the partition
+            # These two are used to control the memory usage of Gemini
+            "token-range-slices": 512,  # Number of partitions
```

Comment:
not sure what's the motivation for all these defaults. and why only 512 partitions